### PR TITLE
Migration to monorepo

### PR DESCRIPTION
**Pull request by ggsdc**
Restructures the whole project to have cornflow-client, cornflow-dags and cornflow-server in the same repository.
Repository should be named cornflow.
We still are missing a main README file.
We should be able to divide part of cornflow-client to a new library, maybe, for example the Instance, Solution, Experiment and Application class.
We could add new libraries that can be useful, such as one with the main components used by cornflow-server on the REST API, so it could be used also by any other REST API that we develop.

We should modify the deployment CI before merging this PR

Closes https://github.com/baobabsoluciones/cornflow-server/issues/231

### CODE REVIEW

**Comment by ggsdc** [comment]
Tests are going to fail (the current ones on the repo, but the new ones should pass without a problem)

File: .github/workflows/test_cornflow_client.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      - name: Install airflow
+        run: |
+          python -m venv afvenv
+          source afvenv/bin/activate
+          AIRFLOW_VERSION=2.2.1
+          PYTHON_VERSION="$(python3 --version | cut -d " " -f 2 | cut -d "." -f 1-2)"
+          CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-${PYTHON_VERSION}.txt"
+          python -m pip install "apache-airflow==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"
+          python -m pip install -U orloge pulp
+          python -m pip install -U git+https://github.com/baobabsoluciones/cornflow-client@master
```
Comment (marioncottard) [inline .github/workflows/test_cornflow_client.yml:88]: I think the url needs to be updated

File: .github/workflows/test_cornflow_client.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
+name: cornflow-client tests
+
+on:
+  pull_request:
+    types: [ opened, edited, synchronize, reopened ]
+    paths:
+      - 'cornflow-client/**'
+      - '!cornflow-clients/README.rst'
```
[the remark this answers is not in the document]
Comment (ggsdc) [reply .github/workflows/test_cornflow_client.yml:8]: You are right

File: .github/workflows/test_cornflow_client.yml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        with:
+          repository: baobabsoluciones/cornflow-server
+          ref: master
+          path: ./cornflow-client/cornflow-server
+      - name: Clone dags from cornflow dags public repo
+        uses: actions/checkout@master
+        with:
+          repository: baobabsoluciones/cornflow-dags-public
+          ref: main
+          path: ./cornflow-client/dags
```
[the remark this answers is not in the document]
Comment (ggsdc) [reply .github/workflows/test_cornflow_client.yml:69]: Will change it.

File: .github/workflows/test_cornflow_client.yml (hunk 1 above)
Comment (ggsdc) [reply .github/workflows/test_cornflow_client.yml:88]: Will change it

**Comment by ggsdc** [comment]
@marioncottard the changes that you requested are done. Are you going to review the PR more? it should only be a migration of the other repositories, I have not made any changes on the code.
@ampuerin do you want to review something?

**Review by marioncottard — APPROVED** [review]
No, I have already finished reviewing it, it looks good to me

**Review by ampuerin — APPROVED** [review]
I think it's all correct! We have to change deployment scripts first.

**Pull request merged**